src/targets/cpu/include/migraphx/cpu/dnnl.hpp

```diff
@@ -300,91 +305,109 @@ struct dnnl_op : auto_register_op<Derived>
     {
         // Compensate for allocation
         inputs.pop_back();
-        const auto& self = static_cast<const Derived&>(*this);
-        auto name        = self.name();
         auto md          = to_memory_desc(output_shape, inputs);
         auto prim        = get_primitive(md);
         auto arg_lookup  = create_arg_map(inputs.size());
+
 #ifndef NDEBUG
-        auto prim_attr = get_primitive_attr(md);
+        execute = std::bind(&dnnl_op::execute_impl_, this, output_shape, inputs, md, prim,
+                            arg_lookup, std::placeholders::_1, std::placeholders::_2);
+#else
+        execute = std::bind(&dnnl_op::execute_impl_, this, md, prim, arg_lookup,
+                            std::placeholders::_1, std::placeholders::_2);
 #endif
-        execute = [=](context&, const std::vector<argument>& args) {
```

Comment:
Assigning a lambda to `std::function` is not UB. This should use the lambda.

Reply:
That is not compiling on Windows. The fatal error message says something like that "cannot assign temporary object (lambda) to class data member".

Reply:
Is this an error on clang?

Reply:
> The fatal error message says something like that "cannot assign temporary object (lambda) to class data member".

Since its complaining about a temporary, maybe you can assign it to a local variable first before assigning it to the member variable.